### PR TITLE
Enhancement: Use sandbox for SDK Testing and remove Indexer v1 steps

### PR DESCRIPTION
**Pull request by algochoi**
- Refactoring make docker-test using pure-shell script test-harness.sh that uses the sandbox instead of the old docker scripts
- Removing steps that belong to features being removed in https://github.com/algorand/algorand-sdk-testing/pull/220

Python SDK PR here: https://github.com/algorand/py-algorand-sdk/pull/367

### CODE REVIEW

File: .test-env
Comment (tzaffi) [inline .test-env:37]:
```suggestion
# WARNING: Be careful when turning on the next variable.
# In that case you'll need to provide all variables expected by `algorand-sdk-testing`'s `.env`
OVERWRITE_TESTING_ENVIRONMENT=0
```

Following a discussion with @michaeldiamant, I ended up dropping most of the variables in [this environment file ](https://github.com/algorand/py-algorand-sdk/blob/c7028689a1ea0af76fdbabc2a6434045b42ae302/.test-env) and opting not to overwrite sdk testing's incoming `.env`. I recommend consistency here, though this is all still under discussion.

File: test-harness.sh
Comment (tzaffi) [inline test-harness.sh:1]: FYI - I've made various changes to the original script based on feedback. The biggest change is [defaulting to not overwriting testing SDK's .env](https://github.com/algorand/py-algorand-sdk/blob/trim-indexer/test-harness.sh#L30-L33)

File: .test-env
Comment (algochoi) [reply .test-env]: updated!

File: test-harness.sh
Comment (algochoi) [reply test-harness.sh:1]: done!

**Review by tzaffi — COMMENTED** [review]
Looking good so far. Still expecting a few more mods (if JS is similar to Python).
## Step removal:
* `then("I get transactions by address only")`
* `then("I can get the transaction by ID")`
* `when("I get recent transactions, limited by {cnt} transactions")`
* `then("I get transactions by address and date")`
## Stop modification:
* `step("I wait for the transaction to be confirmed.")`

**Comment by algochoi** [comment]
@tzaffi  I think I made all the necessary deletions now. For `step("I wait for the transaction to be confirmed.")`, the JS SDK does not support v1 in the first place, so I didn't make any modifications there.

File: tests/cucumber/steps/steps.js
Comment (tzaffi) [inline tests/cucumber/steps/steps.js:619]: I'm surprised this is passing because I still see the step [here](https://github.com/algorand/algorand-sdk-testing/blob/3d2db01121025d8b531d30b83ad1cfa23e5e65c4/features/integration/algod.feature#L22).

**Review by tzaffi — CHANGES_REQUESTED** [review]
I think the [tests aren't running](https://app.circleci.com/pipelines/github/algorand/js-algorand-sdk/448/workflows/639ed9bb-2aac-4d14-a6cf-242f53c3d3cf/jobs/2543), unless I'm missing something?

That would explain how the algod step that I was surprised was removed didn't break the test.

**Comment by tzaffi** [comment]
> I think the [tests aren't running](https://app.circleci.com/pipelines/github/algorand/js-algorand-sdk/448/workflows/639ed9bb-2aac-4d14-a6cf-242f53c3d3cf/jobs/2543), unless I'm missing something?
> 
> That would explain how the algod step that I was surprised was removed didn't break the test.

better link: https://app.circleci.com/pipelines/github/algorand/js-algorand-sdk/448/workflows/639ed9bb-2aac-4d14-a6cf-242f53c3d3cf/jobs/2543?invite=true#step-110-1860

File: tests/cucumber/steps/steps.js
Comment (algochoi) [reply tests/cucumber/steps/steps.js:619]: Good point - I thought we would remove it because it uses the `transactionByAddress` v1 endpoint, but I guess it doesn't invoke the v1 indexer so it should be kept

I'll also try to dig into what is wrong

Comment (algochoi) [reply tests/cucumber/steps/steps.js]: The feature files were being copied to the wrong location - this should be fixed in the [latest commit](https://github.com/algorand/js-algorand-sdk/pull/623/commits/cc0411a834e444f59d1796147d7e1d2c6bed954a)

Comment (algochoi) [reply tests/cucumber/steps/steps.js]: Build: https://app.circleci.com/pipelines/github/algorand/js-algorand-sdk/449/workflows/25e2996f-d027-44af-a2c5-a126832539af/jobs/2547

File: .test-env
Comment (tzaffi) [inline .test-env:4]: Approving with the provision that this PR should wait on the SDK-testing `trim-indexer` branch being merged and then be pointed to `master`.

**Review by tzaffi — APPROVED** [review]
Approving with the provision that this PR should wait on the SDK-testing trim-indexer branch being merged and then be pointed to master.

File: .test-env
Comment (tzaffi) [inline .test-env:6]:
```suggestion
VERBOSE_HARNESS=0

# WARNING: If set to 1, new features will be LOST when downloading the test harness.
# REGARDLESS: modified features are ALWAYS overwritten.
REMOVE_LOCAL_FEATURES=0

# WARNING: Be careful when turning on the next variable.
```

File: test-harness.sh
Comment (tzaffi) [inline test-harness.sh:38]:
```suggestion
## Copy feature files into the project resources
if [[ $REMOVE_LOCAL_FEATURES == 1 ]]; then
  echo "$THIS: OVERWRITE wipes clean tests/features"
  if [[ $VERBOSE_HARNESS == 1 ]]; then
    ( tree tests/features && echo "$THIS: see the previous for files deleted" ) || true
  fi
  rm -rf tests/features
fi
mkdir -p tests/features
cp -r "$SDK_TESTING_HARNESS"/features/* tests/features
if [[ $VERBOSE_HARNESS == 1 ]]; then
  ( tree tests/features && echo "$THIS: see the previous for files copied over" ) || true
fi
```

**Review by tzaffi — COMMENTED** [review]
optional suggestion - I'm defaulting to no longer wiping out the existing features, and introducing a verbose mode

File: test-harness.sh
Comment (tzaffi) [inline test-harness.sh:38]:
```suggestion
ENV_FILE=".test-env"
TEST_DIR="tests/cucumber"

set -a
source "$ENV_FILE"
set +a

rootdir=$(dirname "$0")
pushd "$rootdir"

## Reset test harness
if [ -d "$SDK_TESTING_HARNESS" ]; then
  pushd "$SDK_TESTING_HARNESS"
  ./scripts/down.sh
  popd
  rm -rf "$SDK_TESTING_HARNESS"
else
  echo "$THIS: directory $SDK_TESTING_HARNESS does not exist - NOOP"
fi

git clone --depth 1 --single-branch --branch "$SDK_TESTING_BRANCH" "$SDK_TESTING_URL" "$SDK_TESTING_HARNESS"


if [[ $OVERWRITE_TESTING_ENVIRONMENT == 1 ]]; then
  echo "$THIS: OVERWRITE downloaded $SDK_TESTING_HARNESS/.env with $ENV_FILE:"
  cp "$ENV_FILE" "$SDK_TESTING_HARNESS"/.env
fi

## Copy feature files into the project resources
if [[ $REMOVE_LOCAL_FEATURES == 1 ]]; then
  echo "$THIS: OVERWRITE wipes clean $TEST_DIR/features"
  if [[ $VERBOSE_HARNESS == 1 ]]; then
    ( tree $TEST_DIR/features && echo "$THIS: see the previous for files deleted" ) || true
  fi
  rm -rf $TEST_DIR/features
fi
mkdir -p $TEST_DIR/features
cp -r "$SDK_TESTING_HARNESS"/features/* $TEST_DIR/features
if [[ $VERBOSE_HARNESS == 1 ]]; then
  ( tree $TEST_DIR/features && echo "$THIS: see the previous for files copied over" ) || true
fi
```

File: .test-env
Comment (algochoi) [reply .test-env:12]: done!

**Review by tzaffi — APPROVED** [review]
Re-approving after confirming the newly deleted steps against the expected list.

I'll keep you posted if I discover any new possible deletions.

File: Makefile
Comment (michaeldiamant) [inline Makefile:5]: @algochoi Based on https://github.com/algorand/go-algorand-sdk/pull/360#discussion_r949337599, I think `@indexer.231` ought to be removed.

Note - I didn't assess which additional steps (if any) can also be removed.

```suggestion
	node_modules/.bin/cucumber-js --tags "@algod or @assets or @auction or @kmd or @send or @indexer or @rekey_v1 or @send.keyregtxn or @dryrun or @compile or @applications or @indexer.applications or @applications.verified or @abi or @c2c or @compile.sourcemap" tests/cucumber/features --require-module ts-node/register --require tests/cucumber/steps/index.js
```

Comment (tzaffi) [reply Makefile:5]: I recommend resolving this in favor of the child PR #625

Comment (michaeldiamant) [reply Makefile]: @tzaffi Thanks for the child PR - resolving!

**Review by michaeldiamant — APPROVED** [review]
Approving with the understanding that `.test-env` branch reverted to `master` prior to merge.

File: .test-env
Comment (tzaffi) [inline .test-env:3]:
```suggestion
SDK_TESTING_BRANCH="master"
```
...after the companion SDK Testing PR is merged

**Pull request merged**
